src/plugin/ipc/file/fileconnection.cpp

```diff
@@ -261,7 +261,7 @@ void FileConnection::refill(bool isRestart)
 
   if (!_ckpted_file) {
     int tempfd;
-    if (_type == FILE_DELETED && (_flags & (O_WRONLY | O_RDWR))) {
+    if (_type == FILE_DELETED && ((_flags & O_WRONLY) || (_flags & O_RDWR))) {
```

Comment:
Let's remove `(_flags & O_RDWR)` until we have the full context and commit it. Please create an issue to address 8514737.